### PR TITLE
fix: Make split panel max width calculation deterministic

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Move the code from useEffects into normal code. This prevents layout jitter when resizing split panel or a drawer too quickly, so the effects do not fire on time.

Related links, issue #, if available: AWSUI-22606

### How has this been tested?

* PR build
* Screenshot tests in my pipeline
* Manually to ensure there is no jitter

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/__integ__/app-layout-drawers.test.ts [hunk 1]
```diff
@@ -184,7 +184,7 @@ for (const visualRefresh of ['true', 'false']) {
     test(
       'updates side split panel position when using different width drawers',
       setupTest(
-        { visualRefresh, splitPanelPosition: 'side', screenSize: { ...viewports.desktop, width: 1500 } },
+        { visualRefresh, splitPanelPosition: 'side', screenSize: { ...viewports.desktop, width: 1450 } },
```
Comment (just-boris) [inline src/app-layout/__integ__/app-layout-drawers.test.ts:187]: Reduce the width, to force split panel to the bottom position, as required by the test.

The refactoring change breakpoint when it happens, so the test needs a change

File: src/app-layout/__tests__/split-panel-provider.test.tsx [hunk 2]
```diff
@@ -96,19 +96,14 @@ describe.each(['bottom', 'side'] as const)('position=%s', position => {
       </SplitPanelProvider>
     );
 
-    expect(reportSize).toHaveBeenCalledTimes(2);
+    expect(reportSize).toHaveBeenCalledTimes(position === 'bottom' ? 2 : 1);
```
Comment (just-boris) [inline src/app-layout/__tests__/split-panel-provider.test.tsx:99]: The refactored optimized side position rendering, but not bottom, hence the difference in behavior.

Might do more on fixing the bottom position another time

File: src/app-layout/drawer/resizable-drawer.tsx [hunk 3]
```diff
@@ -18,33 +18,23 @@ import { TOOLS_DRAWER_ID } from '../utils/use-drawers';
...
   refs,
   activeDrawer,
   toolsContent,
   ...props
 }: ResizableDrawerProps) => {
-  const { isOpen, children, isMobile } = props;
-  const [relativeSize, setRelativeSize] = useState(0);
+  const { isOpen, children, width, isMobile } = props;
 
-  useEffect(() => {
```
Comment (just-boris) [inline src/app-layout/drawer/resizable-drawer.tsx:32]: Main subject of this change. Remove `useEffect` calls, make the layout calculated from props directly

File: src/app-layout/index.tsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ import { AppLayoutProps } from './interfaces';
 import { Notifications } from './notifications';
 import { MobileToolbar } from './mobile-toolbar';
 import { useFocusControl } from './utils/use-focus-control';
-import useWindowWidth from './utils/use-window-width';
+import useDocumentWidth from './utils/use-document-width';
```
Comment (just-boris) [inline src/app-layout/index.tsx:14]: `window.innerWidth` includes window scrollbars. So there is a calculation mistake by ~15px when finding available screen space

**Review by gethinwebster — APPROVED** [review]

**Pull request merged**